camille/inputs/bazefetcher.py

```diff
@@ -0,0 +1,41 @@
+from pandas import DataFrame, Timestamp, read_json
+from os import path, walk, listdir
+import dateutil
+
+def _bazefetcher_time_filter(config, filename):
+    filename = path.split(filename)[-1]
+    filename = path.splitext(filename)[0]
+
+    try:
+        s, e = filename.split('_')[-2:]
+    except ValueError:
+        return False
+
+    start = Timestamp(dateutil.parser.parse(s.replace('.',':')))
+    end = Timestamp(dateutil.parser.parse(e.replace('.',':')))
+
+    return config['start'] <= start and end < config['end']
+
+def load(config, base_folder='./'):
+    df = DataFrame()
+    for i, t in enumerate(config['tag']):
+        dirs = [ path.join(base_folder, d)
+                 for d in listdir(base_folder) if  t.match(d) ]
+        files = [ list(map( lambda x: path.join(d, x), listdir(d) ))
+                  for d in dirs ]
+        files = sum( files, [] )
+        files = filter( lambda x: _bazefetcher_time_filter(config, x), files )
+        column = DataFrame()
+        for file in files:
+            f = read_json(file, convert_dates=['t'])
+            f = f.filter(['t','v'])
+            try:
+                f.set_index('t', inplace=True)
+                column = column.append(f[config['start']:config['end']])
+            except KeyError:
+                pass
+
+        df = df.join(column, how=config['join'])
```

Comment:
It is supposedly more performant to collect all the small DataFrames in a list, and then concat them at the end [1], so if you don't find it harder to read (or have any other objections) I propose doing that. 

1: https://tomaugspurger.github.io/modern-4-performance